docs/openapi/howto/binary-payload.md

```diff
@@ -0,0 +1,228 @@
+# How to represent binary/file payload in openapi
+
+This document will explain how you can represent various binary payload in swagger/openapi.
+
+## Basics
+
+There is 2 part to describe a binary/file payload:
+
+- Describe the `content-type`, this could be anything, for example
+
+  - `application/octet-stream`: For a raw binary file.
+  - `image/png`: To describe returning a png file.
+  - `application/json`: To describe returning some json file.(This could be used to describe a json file where the server/sdk has no control over the content)
+
+- Describe the schema of the payload. This is the important part to mark the payload as a binary/file. It is a different process for Swagger 2 and OpenAPI 3
```

Comment:
should we refer to swagger 2 as `OpenAPI 2`?

Reply:
hhm, is that what we have been following?

Reply:
i think so https://github.com/Azure/autorest/tree/master/docs/extensions#autorest-extensions-for-openapi-20